linux_os/guide/services/ssh/ssh_server/firewalld_sshd_port_enabled/bash/shared.sh

```diff
@@ -5,49 +5,47 @@
 
-# Check if any eth interface is bounded to the zone with SSH service enabled
-nic_bound=false
-readarray -t eth_interface_list < <(ip link show up | cut -d ' ' -f2 | cut -d ':' -s -f1 | grep -E '^(en|eth)')
-for interface in "${eth_interface_list[@]}"; do
-    if grep -qi "ZONE=$firewalld_sshd_zone" "{{{ network_config_path }}}"; then
-        nic_bound=true
-        break;
-    fi
-done
-
-if [ $nic_bound = false ];then
-    # Add first NIC to SSH enabled zone
-    interface="${eth_interface_list[0]}"
-
-    if ! firewall-cmd --state -q; then
-        {{% if product in ['rhel9'] %}}
-          {{{ bash_replace_or_append(network_config_path, '^zone=', "$firewalld_sshd_zone", '%s=%s') | indent(8) }}}
-        {{% else %}}
-          {{{ bash_replace_or_append(network_config_path, '^ZONE=', "$firewalld_sshd_zone", '%s=%s') | indent(8) }}}
-        {{% endif %}}
+        # If the connection is not yet assigned to a firewalld zone, assign it to the proper zone.
+        # This will not change connections which are already assigned to any firewalld zone.
+        for connection in "${nm_connections[@]}"; do
+            current_zone=$(nmcli -f connection.zone connection show "$connection" | awk '{ print $2}')
+            if [ $current_zone = "--" ]; then
+                nmcli connection modify "$connection" connection.zone $firewalld_sshd_zone
+            fi
+        done
+        systemctl restart NetworkManager
     else
-        # If firewalld service is running, we need to do this step with firewall-cmd
-        # Otherwise firewalld will communicate with NetworkManage and will revert assigned zone
-        # of NetworkManager managed interfaces upon reload
-        firewall-cmd --permanent --zone="$firewalld_sshd_zone" --add-interface="${eth_interface_list[0]}"
-        firewall-cmd --reload
+        echo "
+        NetworkManager service is not active. Remediation aborted!
+        This remediation could not be applied because it depends on NetworkManager service running.
+        The service is not started by this remediation in order to prevent connection issues."
+        exit 1
     fi
+
+    # Active zones are zones with at least one interface assigned to it.
+    # It is possible that traffic is comming by any active interface and consequently any
+    # active zone. So, this make sure all active zones are permanently allowing SSH service.
+    readarray -t firewalld_active_zones < <(firewall-cmd --get-active-zones | grep -v interfaces)
+    for zone in "${firewalld_active_zones[@]}"; do
+        firewall-cmd --permanent --zone="$zone" --add-service=ssh
+    done
+    firewall-cmd --reload
+else
+    echo "
+    firewalld service is not active. Remediation aborted!
+    This remediation could not be applied because it depends on firewalld service running.
+    The service is not started by this remediation in order to prevent connection issues."
+    exit 1
```

Comment:
Small nitpick: The code might look better if the presence of both firewalld and NM is checked at the beginning because then the rest of the code doesn't have to be inside nested `else` blocks.

Reply:
Correct. Like I did in the Ansible remediation but didn't updated the Bash one. Good. I will do that.

Reply:
Done. Ansible and Bash are now aligned. Thanks for noticing this.

Reply:
OK